src/apps/competitions/tests/test_phase_migration.py

```diff
@@ -0,0 +1,74 @@
+import datetime
+
+from django.test import TestCase
+from django.utils.timezone import now
+
+from competitions.models import Submission
+from competitions.tasks import do_phase_migrations
+from factories import UserFactory, CompetitionFactory, PhaseFactory, SubmissionFactory, SubmissionScoreFactory, \
+    CompetitionParticipantFactory
+
+twenty_minutes_ago = now() - datetime.timedelta(hours=0, minutes=20)
+five_minutes_ago = now() - datetime.timedelta(hours=0, minutes=5)
+
+
+class CompetitionPhaseToPhase(TestCase):
+    def setUp(self):
+        self.owner = UserFactory(username='owner', super_user=True)
+        self.normal_user = UserFactory(username='norm')
+        self.competition = CompetitionFactory(created_by=self.owner)
+        self.competition_participant = CompetitionParticipantFactory(user=self.normal_user,
+                                                                     competition=self.competition)
+        self.phase1 = PhaseFactory(competition=self.competition, auto_migrate_to_this_phase=False,
+                                   end=twenty_minutes_ago, index=1, name='Phase1')
+        self.phase2 = PhaseFactory(competition=self.competition, auto_migrate_to_this_phase=True,
+                                   start=five_minutes_ago, index=2, name='Phase2')
+        self.phase3 = PhaseFactory(competition=self.competition, auto_migrate_to_this_phase=False,
+                                   index=3, name='Phase3')
+
+        for _ in range(4):
+            self.make_submission()
+
+    def make_submission(self, **kwargs):
+        kwargs.setdefault('owner', self.owner)
+        kwargs.setdefault('participant', self.competition_participant)
+        kwargs.setdefault('phase', self.phase1)
+        kwargs.setdefault('status', 'None')
+        sub = SubmissionFactory(**kwargs)
+        SubmissionScoreFactory(submission=sub)
+        return sub
+
+    def test_migrate_submissions(self):
+        assert not self.phase2.submissions.exists()
+        do_phase_migrations()
+        assert self.phase1.submissions.count() == self.phase2.submissions.count()
+
+    def test_currently_migrating_competitions_dont_migrate(self):
+        self.competition.is_migrating = True
+        self.competition.save()
+        do_phase_migrations()
+        assert not self.phase1.submissions.count() == self.phase2.submissions.count()
```

Comment:
another small nitpick, `assert x != y` rather than `assert not x == y`

Reply:
Changed all `assert not x == y` to `assert x !=y`